builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -186,6 +195,7 @@ func resourceCloudStackInstanceRead(d *schema.ResourceData, meta interface{}) er
 	d.Set("display_name", vm.Displayname)
 	d.Set("ipaddress", vm.Nic[0].Ipaddress)
 	d.Set("zone", vm.Zonename)
+	//NB cloudstack sometimes sends back the wrong keypair name, so dont update it
```

Comment:
Hmm... Thinking about it again, let's just keep it as is. It's a simple thing to uodate later when it's behaviour is changed in CS. So will pull this one is shortly (not at a desk rest of this day).